fairlearn/datasets/_fetch_acsincome.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import pathlib
+
+from sklearn.datasets import fetch_openml
+from ._constants import _DOWNLOAD_DIRECTORY_NAME
+
+
+def fetch_acsincome(*, cache=True, data_home=None,
+                    as_frame=False, return_X_y=False):
```

Comment:
Do you want to allow people to filter by state throw an argument?
Or should they do that themselves? The code is a number, not text abbreviation (such as MA), right? Perhaps it's worth copying the mapping over from folktables and map it to the text abbrevation, because nobody is going to understand this.

Reply:
That's correct. Currently, the state code is a number. I agree that's not as convenient though because the numbers are not just 1-51 for the 50 states + Puerto Rico. Should I replace it with the abbreviation and upload a new dataset to openml?

My initial thought was that since this function has to download all of the data anyways (because we're not uploading 51 separate datasets to openml), the function may as well return everything to the user. But let me know if you think we should add a keyword argument for filtering by state(s)

Reply:
I think it's fine on OpenML. I just think it's best to copy the mapping from number to state abbreviation over from folktables and put it in the code. Then add an optional arg where people can pass a list of strings, and by default they get the whole dataset, otherwise just the states they selected (which you can filter down to using the mapping). In any case, I would always drop the state index column before returning to the user, since this is not at all a user-friendly or interpretable piece of information.

Reply:
Added functionality to filter by state. Let me know what you think!